contracts/contracts/Bridge.sol

```diff
@@ -0,0 +1,174 @@
+// contracts/Auth.sol
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+import "./Auth.sol";
+import "./StellarAsset.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+struct WithdrawERC20Request {
+    bytes32 id;
+    uint256 expiration;
+    address recipient;
+    address token;
+    uint256 amount;
+}
+
+struct WithdrawETHRequest {
+    bytes32 id;
+    uint256 expiration;
+    address recipient;
+    uint256 amount;
+}
+
+struct CreateStellarAssetRequest {
+    uint8 decimals;
+    string name;
+    string symbol;
+}
+
+contract Bridge is Auth {
+    constructor(address[] memory _signers, uint8 _minThreshold) Auth(_signers, _minThreshold) {}
+
+    event DepositERC20(
+        address token,
+        address sender,
+        uint256 destination,
+        uint256 amount
+    );
+    event DepositETH(
+        address sender,
+        uint256 destination,
+        uint256 amount
+    );
+
+    // Every bridge transfer has a globally unique id.
+    //
+    // For a ETH -> Stellar transfer the id would be keccak256(abi.encode(txHash, logIndex))
+    // where txHash is the hash of the ethereum transaction containing the deposit event and
+    // logIndex is the index of the deposit event in the ethereum block.
+    // The logIndex is necessary because a single ethereum transaction could call
+    // depositERC20() / depositETH() multiple times.
+    // The bridge smart contract will ensure that a ETH -> Stellar transfer can only be refunded
+    // once by maintaining a set of fulfilled ids.
+    //
+    // For a Stellar -> ETH transfer the id wouold be Stellar transaction hash assuming a
+    // stellar transaction can only contain one Stellar -> ETH transfer.
+    // Similar to the refund case, the bridge smart contract will ensure that a Stellar -> ETH
+    // transfer can only be completed once by mantaining a set of fulfilled ids.
+
+    event WithdrawERC20(
+        bytes32 id,
+        address recipient,
+        address token,
+        uint256 amount
+    );
+    event WithdrawETH(
+        bytes32 id,
+        address recipient,
+        uint256 amount
+    );
+
+    event RegisterStellarAsset(address asset);
+
+    mapping(address => bool) public isStellarAsset;
+
+    function depositERC20(
+        address token,
+        uint256 destination,
+        uint256 amount
+    ) external {
```

Comment:
Should we have the asset code here too or the assumption is to have a separate Bridge contract for each asset?